### PR TITLE
Resize CV-CUDA Backend

### PR DESCRIPTION
**Pull request by justincdavis**
### Summary

Adds the resize CV-CUDA kernel

### Testing

`python3 -m pytest test/test_transforms_v2.py::TestResize`
`python3 -m pytest test/test_transforms_v2.py::TestResize::test_image_correctness_cvcuda`

### CODE REVIEW

File: test/test_transforms_v2.py [hunk 1]
```diff
@@ -862,23 +877,77 @@ def _check_output_size(self, input, output, *, size, max_size):
...
+
+        atol = 1
+        if antialias:
+            # cvcuda.hq_resize is accurate within 9 for the tests
+            atol = 9
+        elif interpolation == transforms.InterpolationMode.BICUBIC:
+            # the CV-CUDA bicubic interpolation differs significantly
+            # importantly, this is only the edge boundaries
+            # visually, there is no noticable difference
+            atol = 91
```
Comment (zy1git) [inline test/test_transforms_v2.py:949]: What is the reason for choosing 91?

File: torchvision/transforms/v2/functional/_geometry.py [hunk 2]
```diff
@@ -401,6 +402,82 @@ def __resize_image_pil_dispatch(
     return _resize_image_pil(image, size=size, interpolation=interpolation, max_size=max_size)
 
 
+_dtype_to_format_cvcuda: dict["cvcuda.Type", "cvcuda.Format"] = {}
+
+
+def _resize_image_cvcuda(
```
Comment (zy1git) [inline torchvision/transforms/v2/functional/_geometry.py:408]: Need to take a closer look at the implementation of this function @NicolasHug 

I looked through the implementation and the logic looks good. However, we might need to:
1, remove some comments.
2, there are some differences between cvcuda interpolation and the pytorch tensor interpolation  in the function `resize_image`(reflected in the parity tests), so we need to be cautious.

File: test/test_transforms_v2.py [hunk 3]
```diff
@@ -862,23 +877,77 @@ def _check_output_size(self, input, output, *, size, max_size):
...
+        atol = 1
+        # when using antialias, CV-CUDA is different for BICUBIC and BILINEAR, since antialias requires hq_resize
+        # hq_resize using interpolation will have differences on the edge boundaries
+        # no noticable visual difference
+        if make_input is make_image_cvcuda and (
+            interpolation is transforms.InterpolationMode.BILINEAR
+            or interpolation is transforms.InterpolationMode.BICUBIC
+        ):
+            atol = 9
+        assert_close(actual, expected, atol=atol, rtol=0)
```
Comment (zy1git) [inline test/test_transforms_v2.py:920]: I am not sure how to verify the `atol = 9`. What is the reason to pick 9?

File: test/test_transforms_v2.py [hunk 4]
```diff
@@ -862,23 +877,77 @@ def _check_output_size(self, input, output, *, size, max_size):
...
+
+    @needs_cvcuda
+    @pytest.mark.parametrize("size", OUTPUT_SIZES)
+    @pytest.mark.parametrize("interpolation", set(INTERPOLATION_MODES) - {transforms.InterpolationMode.NEAREST})
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    @pytest.mark.parametrize("antialias", [True, False])
+    @pytest.mark.parametrize("fn", [F.resize, transform_cls_to_functional(transforms.Resize)])
+    def test_image_correctness_cvcuda(self, size, interpolation, use_max_size, antialias, fn):
```
Comment (zy1git) [inline test/test_transforms_v2.py:928]: I am curious why we have two test_image_correctness functions for cvcuda. If this function test cvcuda specifically, then we don't need to add "make_image_cvcuda" into the "make_ input" for the function in https://github.com/pytorch/vision/pull/9302/changes#diff-9c2dde92db86c123fee225e39b7c1ef96e08a3e79a9dcc9a2d68b21ed51a81d0R896

File: test/test_transforms_v2.py [hunk 5]
```diff
@@ -964,11 +1033,26 @@ def test_keypoints_correctness(self, size, use_max_size, fn):
...
+        ],
     )
     def test_pil_interpolation_compat_smoke(self, interpolation, make_input):
         input = make_input(self.INPUT_SIZE)
 
+        if make_input is make_image_cvcuda and interpolation in {
+            transforms.InterpolationMode.BOX,
+            transforms.InterpolationMode.LANCZOS,
+        }:
+            pytest.skip("CV-CUDA may support box and lanczos for certain configurations of resize")
```
Comment (NicolasHug) [inline test/test_transforms_v2.py]: Should be xfail

File: test/test_transforms_v2.py (hunk 1 above)
Comment (NicolasHug) [reply test/test_transforms_v2.py:949]: > this is only the edge boundaries

QQ Does that mean we can do

```py
        assert_close(actual[:, : , 1:-1, 1:-1], expectedl[:, : , 1:-1, 1:-1])
```

Comment (justincdavis) [reply test/test_transforms_v2.py:949]: IIRC the differences are greatest at the edges of the images, but there are still differences internally in the images regardless. If we use the filtering for the edge values we may find that the max difference is lower, but there will still be one. I believe this behavior is marked as a [known issue on CV-CUDA](https://github.com/CVCUDA/CV-CUDA#:~:text=The%20Resize%20and%20RandomResizedCrop%20operators%20incorrectly%20interpolate%20pixel%20values%20near%20the%20boundary%20of%20an%20image%20or%20tensor%20when%20using%20cubic%20interpolation.%20This%20will%20be%20fixed%20in%20an%20upcoming%20release.) so we could handle it here.

91 was chosen since it was the largest observed difference I would get, but it may make more sense to set a better value like half the max value (as an example).

File: test/test_transforms_v2.py (hunk 3 above)
Comment (justincdavis) [reply test/test_transforms_v2.py:920]: I used 9 since it was the largest difference I observed running tests locally. We should have a better way of picking a threshold.

File: test/test_transforms_v2.py (hunk 4 above)
Comment (justincdavis) [reply test/test_transforms_v2.py:928]: I did this since the existing test compares against PIL, so I made a test which compares CV-CUDA to TorchVision directly. Could be removed if thought to be redundant.

File: test/test_transforms_v2.py (hunk 5 above)
Comment (zy1git) [reply test/test_transforms_v2.py]: done
